### PR TITLE
helm-docs: add migration guides around PodSecurityPolicy and chart 5.0

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Describe how to migrate to Kubernetes 1.25 with the helm chart.
Describe how to migrate from helm chart 4.x to 5.0.

#### Which issue(s) this PR fixes or relates to

Related to: #5357

#### Checklist

- N/A Tests updated
- [X] Documentation added
- N/A `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+1. Choose between the following options:
+
+   1. If you are on Kubernetes version 1.24 and want to keep using PodSecurityPolicy, then merge the following setting into your custom values file:
+
+      ```yaml
+      rbac:
+        forcePSPOnKubernetes124: true
+      ```
+
+      > **Note**: Warning: this value prevents you from upgrading to Kubernetes version 1.25! Follow the other option before upgrading to Kubernetes 1.25.
```
Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md:28]: Please revise per https://grafana.com/docs/writers-toolkit/writing-guide/shortcodes/#admonition-shortcode.

Comment (osg-grafana) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md:28]:
```suggestion
      > **Note**: Warning: this value prevents you from upgrading to Kubernetes version 1.25. Follow the other option before upgrading to Kubernetes 1.25.
```

**Review by osg-grafana — APPROVED** [review]
See comment about note and warning admonitions.

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-to-kubernetes-version-1.25.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Migrate to Kubernetes version 1.25
+
+This procedure describes how to prepare a Grafana Mimir Helm chart deployment for an upgrade to Kubernetes 1.25.
+
+## Background
+
+Kubernetes version 1.25 removes the support for the deprecated [PodSecurityPolicy](https://kubernetes.io/docs/concepts/security/pod-security-policy/) object. You can learn more about this topic by visiting [PodSecurityPolicy Deprecation: Past, Present, and Future](https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/).
+
+Due to how Helm [works](https://helm.sh/docs/topics/kubernetes_apis/), PodSecurityPolicy objects must already be removed from the deployment on Kubernetes version 1.24. If you are using PodSecurityPolicy on Kubernetes 1.24, this is a breakind change. Grafana Mimir helm chart versions before 5.0 used PodSecurityPolicy by default.
```
Comment (dimitarvdimitrov) [inline docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-to-kubernetes-version-1.25.md:16]:
```suggestion
Due to how Helm [works](https://helm.sh/docs/topics/kubernetes_apis/), PodSecurityPolicy objects must already be removed from the release prior to upgrading to Kubernetes version 1.25. If you are using PodSecurityPolicy on Kubernetes 1.24, this is a breaking change. Grafana Mimir helm chart versions before 5.0 used PodSecurityPolicy by default in Kubernetes version 1.24.
```

* Isn't "release" a more accurate term? I think that's what each different installation is called


* Also technically it shouldn't be removed in 1.24, but on whatever release was prior to the upgrade to 1.25.
* Also last point about making it clear what the behaviour was - it was by default in <=1.24

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Grafan Mimir Helm chart version 5.0 and later does not install [PodSecurityPolicy](https://kubernetes.io/docs/concepts/security/pod-security-policy/) objects on Kubernetes version 1.24 by default to prepare for upgrade to Kubernetes version 1.25.
+
+## Prerequisite
+
+- This procedure is only applicable if `rbac.create` is `true` and `rbac.type` is `psp` in your current Helm values. This was the case by default before Helm chart version 5.0.
+
+## Procedure
+
+1. If `rbac.create` is `false` or `rbac.type` is `scc`, then there is nothing to do, skip the whole procedure.
```
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md:18]: same question applies to the other article, which has the same first step

**Review by dimitarvdimitrov — APPROVED** [review]
Thanks for writing. Only a few minor suggestions, but otherwise LGTM

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (krajorama) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md:18]: yes,  I don't really trust people reading everything

File: docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+1. Choose between the following options:
+   1. If you are on Kubernetes version 1.24 and want to keep using PodSecurityPolicy, then merge the following setting into your custom values file:
+      ```yaml
+      rbac:
+        forcePSPOnKubernetes124: true
+      ```
+   1. If you have Kubernetes version 1.22 or later and want to upgrade to Kubernetes version 1.25 after this procedure then follow the [Migrate to Kubernetes version 1.25]({{< relref "./migrate-to-kubernetes-version-1.25.md" >}}) guide.
```
[the remark this answers is not in the document]
Comment (krajorama) [reply docs/sources/helm-charts/mimir-distributed/migration-guides/migrate-helm-chart-4.x-to-5.0.md:25]: Oh, I didn't see this due to auto merge. Yes, I suppose that's valid, just have to formulate carefully.

**Pull request merged**